src/screens/HyperSwitch/PaymentLogs/PaymentLogs.res

```diff
@@ -323,139 +362,165 @@ let make = (~paymentId, ~createdAt) => {
-    | _ => PageLoaderWrapper.Loading
+  let screenState = React.useMemo1(() => {
+    logs.contents = logs.contents->Js.Array2.sortInPlaceWith(sortByCreatedAt)
+
+    switch logs.contents->Array.get(0) {
+    | Some(value) => {
+        let initialData = value->getDictFromJsonObject
+        switch initialData->getLogType {
+        | PAYMENTS => {
+            let request = initialData->getString("request", "")
+            let response = initialData->getString("response", "")
+            setLogDetails(_ => {
+              response,
+              request,
+            })
+            setSelectedOption(_ => {
+              value: initialData->getString("request_id", ""),
+              optionType: PAYMENTS,
+            })
+          }
+        | SDK => {
+            let request =
+              initialData
+              ->Dict.toArray
+              ->Array.filter(entry => {
+                let (key, _) = entry
+                filteredKeys->Array.includes(key)->not
+              })
+              ->Dict.fromArray
+              ->Js.Json.object_
+              ->Js.Json.stringify
+            let response =
+              initialData->getString("log_type", "") === "ERROR"
+                ? initialData->getString("value", "")
+                : ""
+            setLogDetails(_ => {
+              response,
+              request,
+            })
+            setSelectedOption(_ => {
+              value: initialData->getString("event_id", ""),
+              optionType: SDK,
+            })
+          }
+        | WEBHOOKS => {
+            let request = initialData->getString("outgoing_webhook_event_type", "")
+            let response = initialData->getString("content", "")
+            setLogDetails(_ => {
+              response,
+              request,
+            })
+            setSelectedOption(_ => {
+              value: initialData->getString("event_id", ""),
+              optionType: WEBHOOKS,
+            })
+          }
+        }
+      }
+    | _ => ()
     }
-  }, (screenState1, screenState2))
+
+    screenState
+  }, [screenState])
 
   React.useEffect0(() => {
     getDetails()->ignore
     None
   })
 
+  open OrderUtils
   <PageLoaderWrapper
-    screenState customUI={<NoDataFound message="No logs available for this payment" />}>
-    {if paymentId->HSwitchOrderUtils.isTestPayment {
+    screenState customUI={<NoDataFound message="No logs available for this PAYMENTS" />}>
+    {if paymentId->HSwitchOrderUtils.isTestPayment || logs.contents->Array.length === 0 {
       <div
         className="flex items-center gap-2 bg-white w-full border-2 p-3 !opacity-100 rounded-lg text-md font-medium">
         <Icon name="info-circle-unfilled" size=16 />
         <div className={`text-lg font-medium opacity-50`}>
-          {"No logs available for this payment"->React.string}
+          {"No logs available for this PAYMENTS"->React.string}
```

Comment:
`{"No logs available for this PAYMENTS"->React.string}`


It should be Payment only i guess. Only one payment we are dealing at a time I guess